kv/kvclient/kvcoord: TestAsyncAbortPoisons failed

kv/kvclient/kvcoord.TestAsyncAbortPoisons [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3532165&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3532165&tab=artifacts#/) on master @ [bc42b6626366da8b2cdd470382dd5fb8202ba0c5](https://github.com/cockroachdb/cockroach/commits/bc42b6626366da8b2cdd470382dd5fb8202ba0c5):


```
=== RUN   TestAsyncAbortPoisons
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAsyncAbortPoisons372585733
    test_log_scope.go:80: use -show-logs to present logs inline
    dist_sender_server_test.go:1516: expected TransactionAbortedError\(ABORT_REASON_ABORT_SPAN\), got: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "unnamed" meta={id=d5d0e84e key="a" pri=0.01373277 epo=0 ts=1633241972.682515576,0 min=1633241972.682515576,0 seq=1} lock=true stat=PENDING rts=1633241972.682515576,0 wto=false gul=1633241973.182515576,0
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAsyncAbortPoisons372585733
--- FAIL: TestAsyncAbortPoisons (7.95s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestAsyncAbortPoisons PKG=./pkg/kv/kvclient/kvcoord TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/kv andreimatei
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAsyncAbortPoisons.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go
@@ -14,7 +14,6 @@ import (
 	"bytes"
 	"context"
 	"fmt"
-	"regexp"
 	"sort"
 	"strings"
 	"sync/atomic"
@@ -1472,31 +1471,26 @@ func TestAsyncAbortPoisons(t *testing.T) {
 					if r.Poison {
 						close(commitCh)
 					} else {
-						commitCh <- fmt.Errorf("EndTxn didn't have expected Poison flag")
+						commitCh <- errors.New("EndTxn didn't have expected Poison flag")
 					}
 				}
 			}
 		}
 		return nil
 	}
-	s, _, _ := serverutils.StartServer(t,
+	s, _, db := serverutils.StartServer(t,
 		base.TestServerArgs{Knobs: base.TestingKnobs{Store: &storeKnobs}})
 	ctx := context.Background()
 	defer s.Stopper().Stop(ctx)
-
-	// Setup two userspace ranges: /Min-b, b-/Max.
-	db := s.DB()
 
 	// Write values to key "a".
 	txn := kv.NewTxn(ctx, db, 0 /* gatewayNodeID */)
 	b := txn.NewBatch()
 	b.Put(keyA, []byte("value"))
-	if err := txn.Run(ctx, b); err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, txn.Run(ctx, b))
 
 	// Run a high-priority txn that will abort the previous one.
-	if err := db.Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+	require.NoError(t, db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		if err := txn.SetUserPriority(roachpb.MaxUserPriority); err != nil {
 			return err
 		}
@@ -1507,17 +1501,13 @@ func TestAsyncAbortPoisons(t *testing.T) {
 			return err
 		}
 		return txn.Put(ctx, keyA, []byte("value2"))
-	}); err != nil {
-		t.Fatal(err)
-	}
+	}))
 
-	expErr := regexp.QuoteMeta("TransactionAbortedError(ABORT_REASON_ABORT_SPAN)")
-	if _, err := txn.Get(ctx, keyA); !testutils.IsError(err, expErr) {
-		t.Fatalf("expected %s, got: %v", expErr, err)
-	}
-	if err := <-commitCh; err != nil {
-		t.Fatal(err)
-	}
+	_, err := txn.Get(ctx, keyA)
+	require.Error(t, err)
+	require.IsType(t, err, &roachpb.TransactionRetryWithProtoRefreshError{})
+	require.Contains(t, err.Error(), "TransactionAbortedError")
+	require.NoError(t, <-commitCh)
 }
 
 // TestTxnCoordSenderRetries verifies that the txn coord sender

